### DISCUSSION TITLE
Dealing with complex spatio-temporal sampling design/time-varying samplers argument?

### DISCUSSION THREAD

**Discussion by adietzel** [inlabru-org/inlabru#135] [Q&A] [answered]
Dear inlabru community,

I would like to use inla(bru) to model the distribution of species in the city of Zurich as a function of a bunch of covariates that vary in space (e.g. % green space, distance to water) or space AND time (climate variables like temperature). Due to resource constraints ecological surveys (geo-referenced observations of species) were conducted suburb-by-suburb over the course of a decade. Each suburb will be surveyed a second time over the next decade to assess temporal changes in species assemblages. 

For most covariates I only have spatial data for a single snapshot in time, which is both inevitable given the cost of such data and sufficient given how slowly these variables change through time. Climate variables, however, exhibit at least as much temporal as spatial variance at the scale of a city like Zurich and I am unsure how to incorporate the complex spatio-temporal nature of my input data in a statistical model. The two ideas I have come up with so far are:

a) To model time implicitly rather than explicitly by collapsing the temporal dimension and running a spatial-only model. This would lead to sharp transitions in climate variables at suburb boundaries and I am unsure how useful and interpretable the modeled spatial latent field would be. Could I use a barrier spde model to indicate suburb boundaries as barriers in the spatial point process to make more sensible inference?

b) Model both space AND time explicitly. I would suspect that the model would be "confused" (apologies for lack of a better word) by the plot or suburb-wise sampling of the response variable unless explicitly told so. I have come across the samplers argument used for example in the gorilla example of Bachl et al. (2019, MEE) to indicate which parts of the study region were surveyed.
Would it be possible to parse multiple (one per year) SpatialPolygonsDataFrames to the 'samplers' argument to indicate which parts of the study region/suburbs were surveyed in a given year?

I am new to R-INLA and inlabru and would very much appreciate any guidance including any references to papers or books. 

Best wishes and thanks for your help.

Andreas

**Comment by Janineillian** [comment]
Hi Andreas,
thanks for your message. It looks to me as if it would be easier to discuss your questions in a meeting rather than through the forum here as you are probably looking for both some modelling advice and technical support. 
Would you be interested in a chat with me (and perhaps others) to point you in the right direction?

Cheers

Janine

**Reply by ASeatonSpatial** [reply]
I'd be happy to discuss as Janine suggests.  A couple of questions though:

Is your data a point process?  The samplers argument is a specific option for the Cox process likelihood, used to define appropriate numerical integration schemes.  See `?like` and `?lgcp`.  

If you don't have a point process and instead have e.g. data at an aggregated level (like to regular grid or administrative areal unit), then the data could simply be the collection of polygons / grid cells with the relevant temporal information included as covariate values.  

Things get more complicated if you want a spatio-temporal random field (although if only one suburb is surveyed each year (and these are not close in space) then there might not be much temporal correlation in the data to inform a spatio-temporal field?).

**Reply by adietzel** [reply]
Hi Janine and Andy,
Thank you for your replies and offer to help. It would be great to have a chat about our project with either or both of you. I am currently in Australia but happy to be flexible to find a time that works for everyone. 

I likely don't yet fully understand the difference between an LGCP and a grid-based analysis but I believe I could do either. I am drawing inspiration for my study from a paper by Adde et al. (2020, https://onlinelibrary.wiley.com/doi/full/10.1111/ddi.13129). Their study seems to be grid based though they are using the centroids of each grid cell as the location of observations and are using an spde model to model the spatial latent field. I have GPS coordinates for all observations but am planning to convert those into a presence grid with a grid cell size of 25x25m. All covariates will be calculated across this grid as well. In the code snippet Adde et al. provide, they group observations by year:

# define SPDE
spde <- inla.spde2.matern(mesh = mesh, alpha = 2)

# create indices
groups <- ducks$year
ducks$ind <- group_indices(ducks@data, year)

# set formula
form.bru <- as.formula(
  paste(response, paste(c('Intercept', covariates, 
  'mySPDE(main = coordinates, ngroup = length(unique(groups)), 
  group = ind,  model = spde)'), collapse=" + "), sep='~'))

Is this equivalent to the collection of grid cells you (Andy) are suggesting as a work-around? Since Adde et al. only included grid cells where observations were available for at least 5 years but up to 25 years of observations for some grid cells, they also dealt with a spatio-temporal sampling design which is not spatially consistent between years, however not to the same extent as in my case. 

I am not necessarily interested in temporal changes but climatic conditions need to correspond to the climate of the year the observation was made and I am unsure how the spatial latent field will deal with sharp transitions in covariate values at suburb boundaries. I hope this is somewhat more clear. 

Thanks a lot for your help. 
Best

Andreas

**Reply by adietzel** [reply]
I also just rediscovered the tutorials on your website. In Tutorial 8 (spatio-temporal), year is also used as a grouping variable. I presume that's the best path forward for me at this stage. I would still be happy to chat with either of you two but I am also wary of using your precious time at a stage when I clearly have to do get better acquainted with INLA and inlabru. Thanks so much for your help!
Best wishes

Andreas

**Reply by finnlindgren** [reply]
Make sure you look at the tutorial/example versions on [https://inlabru-org.github.io/inlabru/articles/](https://inlabru-org.github.io/inlabru/articles/) and not the out-of-date versions on inlabru.org

**Comment by Janineillian** [comment]
Hi Andreas,
good to see that things are becoming clearer to you. Just let us know if you feel you'd like to have chat!

Cheers

Janine






Prof. Janine Illian

Chair in Statistical Science


School of Mathematics and Statistics
The Mathematics and Statistics building


University Place
University of Glasgow
Glasgow G12 8QQ

Direct line: +44(0)1413304749<tel:+44(0)1413304749>
Fax: +44 (0)141 330 4814<tel:+44%20(0)141%20330%204814>

The University of Glasgow, charity number SC004401

________________________________

**Comment by finnlindgren** [comment, marked as answer]
To add just some brief comments on two of your specific technical questions:

- "Could I use a barrier spde model to indicate suburb boundaries as barriers in the spatial point process to make more sensible inference?".
Aside from the modelling aspect, yes, you can use barrier models with inlabru. You need to specify a mapper to the mesh, by adding `mapper=bru_mapper(mesh)` to the component definition.

- "Would it be possible to parse multiple (one per year) SpatialPolygonsDataFrames to the 'samplers' argument to indicate which parts of the study region/suburbs were surveyed in a given year?"
Yes, this would work the same as for space-time distance sampling models, where each line transect would have its own time information as a column in the data frame. If the column is "year", then you'd need `domain = list(coordinates=mesh, year=the_ordered_set_of_years)`. The year domain specification will depend on how you're mapping years in the model component; it needs to match. In plain INLA, one has to use integers starting from 1, but in inlabru there's the option to use actual years to define the component mappings.

**Reply by adietzel** [reply]
Thank you for your reply!
